purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -1202,15 +1202,21 @@ class Purchases internal constructor(
     ) {
         purchases.forEach { purchase ->
             if (purchase.purchaseState != PurchaseState.PENDING) {
-                if (purchase.purchaseOptionId != null) {
+                purchase.productIds.firstOrNull()?.let { productId ->
                     billing.queryProductDetailsAsync(
                         productType = purchase.type,
-                        productIds = purchase.productIds.toSet(),
+                        productIds = setOf(productId),
```

Comment:
Hmm is there any reason for this change? I think purchases should only have a single productId for now (until multi-line purchases maybe?) but we can still query all productIDs and just use the first store product inside the completion block, same as before. I might be missing something though.

Reply:
Ummm... I think you right! I'll revert this since this could get missed in the future when we do need to support multiple

Reply:
i think it might be safer to get the single product Id here, otherwise we query for multiple store products. hopefully we only pass the single ID higher up in the call chain, but this is where it really matters

Reply:
though that is a functionality change so i'm okay if we want to avoid those for now...

Reply:
Well, right now it shouldn't really matter, since purchases should only have a single product ID. But I guess I would prefer to minimize the changes unless there is a good reason. 

Also, if devs start using multi-line purchases without proper support, I wouldn't focus in optimizing, but rather on making sure they know that's not supported. As in, if we detect a purchase with multiple product ids, I would prefer to fail rather than just pick the first and continue the process, and we should fail as early as possible.

Reply:
okay sure that makes sense to me. if we go that route maybe we can make a new ticket to handle those failures